test/js/loan_store_test.js

```diff
@@ -96,25 +364,240 @@ describe('Loan store tests', function() {
 
     });
 
- 
     describe('fetch loan data', function() {
 
+        it ('should not needed to stop request when no current request, and will update calculated properties when new request finishes successfully',
```

Comment:
Not really, I am just spelling out the paths it took within the function and what are things being called.  There is only 1 unit under test.